### PR TITLE
Feat#82/fetch credentials to do crud on third party tools

### PR DESCRIPTION
**Pull request by MounikaReddy15**
## Purpose
_Fetch credentials to do crud on third party tools_

## Further info
_All crud operations are working on insights, trello & github_

## Ticket number
#82

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -33,6 +34,7 @@ const App = () => (
           />
           <Route path={routes.LOGIN} component={Login} />
           <Route path={routes.REGISTER} component={Register} />
+          <Route path={routes.MISSING_DATA} component={MissingData} />
```
Comment (RadhikaPPatel) [inline src/App.js:37]: Why is this page moved here? Anything that needs to be accessed post-login needs to be in Container. Anything that needs to be accessed without login will be in App.

File: src/layout/TopBar/TopBar.js [hunk 2]
```diff
@@ -67,14 +67,6 @@ const TopBar = ({
     setOrganization(user.organization.name);
   }
 
-  useEffect(() => {
```
Comment (RadhikaPPatel) [inline src/layout/TopBar/TopBar.js:70]: I guess you will need this back as well.

File: src/pages/Dashboard/components/UserDashboard.js [hunk 3]
```diff
@@ -150,7 +153,7 @@ const UserDashboard = (props) => {
 
     setProductFeatures(_.orderBy(feats, 'create_date', 'desc'));
     setProductIssues(_.orderBy(iss, 'create_date', 'desc'));
-  }, [product, features, issues]);
+  }, [product, features, issues, features.length, issues.length]);
```
Comment (RadhikaPPatel) [inline src/pages/Dashboard/components/UserDashboard.js:156]: Remove these from here. Do not length check in useEffect run condition. Instead use it inside useEffect.

File: src/pages/Dashboard/components/UserDashboard.js [hunk 4]
```diff
@@ -213,36 +216,58 @@ const UserDashboard = (props) => {
     const { id, type } = toDeleteItem;
     setDeleteModal(false);
     if (type === 'feat') {
-      dispatch(deleteFeature(id));
+      const featCred = _.find(
```
Comment (RadhikaPPatel) [inline src/pages/Dashboard/components/UserDashboard.js:219]: Why do we need to fetch the creds here again? Can't we just pass it in the callback?

**Review by RadhikaPPatel — CHANGES_REQUESTED** [review]
Please cross-check all changes once again as well as retest with dev BE once.

**Comment by yasmin2496** [comment]
Unable to convert to issue and create new issue as well. Please check @MounikaReddy15 @mthombare . Returns 400
image

**Review by yasmin2496 — CHANGES_REQUESTED** [review]
Issue creation is not working

**Comment by MounikaReddy15** [comment]
> Issue creation is not working

@yasmin2496 I checked every CRUD is working fine

**Review by yasmin2496 — CHANGES_REQUESTED** [review]
1. Can the Kanban view have less padding space. Instead of using padding as 16px could you check with 8px. Right now, it looks like there is good amount of empty space left
2. In Kanban view, if you move the card in any of the columns it is not reflected on the source 
3. For issue, update made to Organization is not retained. It clears out everytime issue details is opened

**Review by yasmin2496 — APPROVED** [review]
Suggestions going in ticket - https://github.com/buildly-release-management/buildly-react-template/issues/89

**Review by RadhikaPPatel — CHANGES_REQUESTED** [review]
Requested changes for below

1. Feature not being created on GitHub Project board. Is it FE or BE issue?
2. If we change the Feature Status say from Backlog to In Progress, how does it move the corresponding cards in Trello or GitHub to the correct columns/swimlanes?

**Review by RadhikaPPatel — APPROVED** [review]
2 points notified will be covered in a separate PR

**Review by yasmin2496 — APPROVED** [review]
Looks good

**Pull request merged**